test.py

```diff
@@ -29,12 +27,25 @@ def test_get_soc_code(self, mocked_request):
         mocked_request.assert_called_with('https://api.lmiforall.org.uk/api/v1/soc/search?q=1', verify=False)
         self.assertEqual(expected_data, actual_data)
 
+    @mock.patch('requests.get')
+    def test_invalid_job_input(self, mocked_request):
+        job = "invalid_job"
+        expected_data = job + " is not a valid input"
+        mocked_request.return_value = Mock(text='[]', status_code=200)
+        try:
+            actual_data = main.get_soc_code(job)
+        except Exception as e:
+            actual_data = str(e)
+
+        mocked_request.assert_called_with('https://api.lmiforall.org.uk/api/v1/soc/search?q=invalid_job', verify=False)
+        self.assertEqual(expected_data, actual_data)
```

Comment:
You could use this as a better way to check the exception 
https://stackoverflow.com/questions/129507/how-do-you-test-that-a-python-function-throws-an-exception

Reply:
Updated